translate/yaml.go

```diff
@@ -0,0 +1,90 @@
+package translate
+
+import (
+	"fmt"
+
+	"github.com/arr-ai/arrai/rel"
+	"gopkg.in/yaml.v2"
+)
+
+// YAMLToArrai translates an object unmarshalled from yaml into an array value
+//
+// translation follows the rules
+//
+//     object -> {|@,@item|, |key,val|, ...}
+//     array  -> array
+//     null   -> none
+//     other  -> value (bools, numerics, strings)
+
+func YAMLToArrai(data interface{}) (rel.Value, error) {
+	switch v := data.(type) {
+	case map[interface{}]interface{}:
+		{
+			mapString := make(map[string]interface{})
+
+			for key, value := range v {
+				strKey := fmt.Sprintf("%v", key)
+				mapString[strKey] = value
+			}
+			return yamlObjToArrai(mapString)
+		}
+	case []interface{}:
+		return yamlArrToArrai(v)
+	case string:
+		return rel.NewTuple(rel.NewAttr("s", rel.NewString([]rune(v)))), nil
+	case float64:
+		return rel.NewNumber(v), nil
+	case bool:
+		return rel.NewTuple(rel.NewAttr("b", rel.NewBool(v))), nil
+	case nil:
+		return rel.NewTuple(rel.NewAttr("null", rel.None)), nil
+	default:
+		t, err := rel.NewValue(v)
+		if err != nil {
+			return nil, err
+		}
+		return rel.NewTuple(rel.NewAttr("v", t)), nil
+	}
+}
```

Comment:
There's a similar function that I wrote in `translate/json.go`, you should use that to avoid duplicate.

Reply:
No problems, I'll refactor that to be a helper for both.